### PR TITLE
Tons of Grinch bugfixes and polish

### LINKED ISSUE #21136: The Grinch:tm: can't strip items off people
(WEB REPORT BY: coxdez REMOTE: 206.221.180.138:7777)
# Revision
82f05ba7c72c4337598d86aa6d0b9e7779f7b38f

# Description
As the title say, the Grinch can't strip anything from people despite having hands.

# Steps to Reproduce
Be a shitter and get a flash, then try to strip a stunned person, or just try to strip someone who's braindead
# What you Expected
Stripping

# What Actually Happened
No stripping and disappointment

### LINKED ISSUE #34597: Grinch antag can't be forced in the game panel
(WEB REPORT BY: chazach REMOTE: 206.221.180.138:7777)
# Revision
35673e3083f35c5db7d06548612a5dfbb776c9ab

# Description
ADMIN LOG: DYNAMIC MODE: The ruleset couldn't be executed due to lack of eligible players. (not true there were 19 eligible)
Also doesnt work as the roundstart one
# What you Expected
grinch funnies
# What Actually Happened
NO grinch EVER

### PR DESCRIPTION
**Pull request by DeityLink**
image

Fixes #34597
Fixes #21136

:cl:
* bugfix: Fixed the Grinch rulesets being unable to fire outside of Christmas season even when forced by admins.
* bugfix: Fixed the roundstart Grinch ruleset never working.
* bugfix: Fixed countless Grinch inventory-related issues, such as items disappearing from your inventory if you close and re-open the game, being unable to un-equip or re-equip your backpack, having an unused uniform slot, being unable to strip crew members or getting stripped yourself, etc...
* bugfix: Fixed Santa's Bag being able to carry way less items than it was supposed to. The Grinch's bag is now a sub-type of santa's bag.
* imageadd: The Grinch now has its own logo for the scoreboard.
* imageadd: The Grinch bag now looks like the wizard's Santa bag. Both bags now also visually grow larger as you fit more items in them, and both bags now have in-hand icons.

### CODE REVIEW

**Comment by Chazach** [comment]
merge this.....NOW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -402,17 +402,17 @@
 		src << browse(null, t1)
 
 	if(href_list["hands"])
-		if(usr.incapacitated() || !Adjacent(usr)|| isanimal(usr))
+		if(usr.incapacitated() || !Adjacent(usr)|| (isanimal(usr) && !isgrinch(usr)))
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/carbon.dm:405]: the amount of staggeringly ugly exceptions we're making for grinch here begs the question: is there ANY reason we should not just upgrade the grinch to at least a carbon that simply has a gremlin act spell? this way he would have REAL HANDS and not need constant exceptions for being an animal

Comment (DeityLink) [reply code/modules/mob/living/carbon/carbon.dm:405]: I asked that exact question on Discord when I was working on it and I agree with that sentiment, but I also really spent more efforts on that PR than I thought I would (how hard can it be to fix the Grinch I cluelessly told myself) and I'd like not to turn it into a carbon in this PR because I can forsee another all nighter to test and fix all the edge cases. If it's necessary to have the PR get merged then I'll let it sleep until I find the motivation to tackle the change properly.

File: code/datums/gamemode/role/grinch.dm [hunk 2]
```diff
@@ -27,7 +30,7 @@
 		return
 
 	var/icon/logo = icon('icons/logos.dmi', logo_state)
-	to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are the Grinch!</span><span class='warning'>You are here to ruin Christmas!</span>")
+	to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are the Grinch!</span> <span class='warning'>You are here to ruin Christmas by stealing as many things into your bag as possible! You can crawl through vents and mess with electronic panels and computers, but your pitiful punches don't deal any damage.</span>")
```
Comment (Hinaichigo) [inline code/datums/gamemode/role/grinch.dm]: I kinda prefer the pre-change text here. Short, sweet, and ambiguously open-ended.

Comment (DeityLink) [inline code/datums/gamemode/role/grinch.dm]:
```suggestion
	to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are the Grinch!</span> <span class='warning'>You are here to ruin Christmas! Your bag can hold a ludicrous amount of items. You can crawl through vents and mess with electronic panels and computers, but your pitiful punches don't deal any damage.</span>")
```

Comment (DeityLink) [reply code/datums/gamemode/role/grinch.dm]: I changed the wording to so it retains its initial ambiguity, while still providing information on the Grinch's abilities.

**Comment by DeityLink** [comment]
Fixed conflicts
